src/hooks/transactions/helpers/setTransactionNonces.ts

```diff
@@ -0,0 +1,22 @@
+import { Transaction } from '@multiversx/sdk-core';
+
+export const setTransactionNonces = (
+  latestNonce: number,
+  transactions: Array<Transaction>
+): Array<Transaction> => {
+  if (transactions.length === 0) {
+    return transactions;
+  }
+
+  return transactions.map((tx: Transaction, index: number) => {
+    const nextNonce = latestNonce + index;
+
+    const txNonce = tx.getNonce().valueOf();
+    // stop replacing nonce if transaction is configured with a higher nonce than the existing one
+    const computedNonce = txNonce > nextNonce ? txNonce : nextNonce;
```

Comment:
just a suggestion: maybe create a small helper called `getTransactionNonce` and use it here and in newTransaction so we isolate this logic in one place